### PR TITLE
Monorepo

### PR DESCRIPTION
**Pull request by quietbits**
- Added workspaces `demo-wallet-client`, `demo-wallet-server`, and `demo-wallet-shared` (in `packages`)
- Moved `constants`, `helpers`, and `methods` to shared
- Check root `package.json` for scripts to run for development and builds

Currently shared code can be imported as shown below. I'm not a huge fan of having to specify `/build` (we need to use transpiled JS code instead of TS), but it works for now. We can spend more time later to make this better.

```typescript
import {...} from "demo-wallet-shared/build/...;
```

1) demo-wallet-server will act as a backend server for sep-10 client attribution (transaction signing) and also serve a stellar.toml file. 
2) during the sep-10 flow, the wallet client will request a challenge from the anchor server, that challenge gets passed to the wallet server for signing (client attribution) and returned to the wallet client. 
3) the wallet client will then sign that challenge before sending the challenge transaction back to the anchor server for verification. 

now seen in the sep-10 flow:
image

TODO: add diagram

### CODE REVIEW

**Comment by tprobinson** [comment]
PR previews will still be broken until I get Jenkins and Kube merged. In the meantime feedback welcome on anything I've changed.

File: Dockerfile-client [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+MAINTAINER SDF Ops Team <ops@stellar.org>
+
+WORKDIR /app
+
+ARG REACT_APP_AMPLITUDE_KEY
+ENV REACT_APP_AMPLITUDE_KEY $REACT_APP_AMPLITUDE_KEY
+
+ARG REACT_APP_SENTRY_KEY
+ENV REACT_APP_SENTRY_KEY $REACT_APP_SENTRY_KEY
```
Comment (piyalbasu) [inline Dockerfile-client:11]: could you actually rm the references to `REACT_APP_SENTRY_KEY` and `REACT_APP_AMPLITUDE_KEY`? These are no longer used

File: Makefile [hunk 2]
```diff
@@ -4,13 +4,23 @@ SUDO := $(shell docker version >/dev/null 2>&1 || echo "sudo")
...
 
-docker-push:
-	$(SUDO) docker push $(TAG)
+docker-push-server:
+	$(SUDO) docker push $(SERVER_TAG)
+
+docker-build-client:
+	$(SUDO) docker build -f Dockerfile-client --pull --label org.opencontainers.image.created="$(BUILD_DATE)" \
+	--build-arg REACT_APP_AMPLITUDE_KEY=$(AMPLITUDE_KEY) --build-arg REACT_APP_SENTRY_KEY=$(SENTRY_KEY) \
+	--build-arg REACT_APP_CLIENT_DOMAIN=$(REACT_APP_CLIENT_DOMAIN) --build-arg REACT_APP_WALLET_BACKEND_ENDPOINT=$(REACT_APP_WALLET_BACKEND_ENDPOINT) \
```
Comment (piyalbasu) [inline Makefile:22]: same here - we don't need these build arg's for` REACT_APP_AMPLITUDE_KEY` and `REACT_APP_SENTRY_KEY` so let's get rid of em

File: packages/demo-wallet-server/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}));
+
+// Sign requests from the demo wallet client for sep-10 client attribution
+app.post("/sign", (req, res) => {
+    console.log("request to /sign");
+    const envelope_xdr = req.body.transaction;
+    const network_passphrase = req.body.network_passphrase;
+    const transaction = new Transaction(envelope_xdr, network_passphrase);
+
+    if (Number.parseInt(transaction.sequence, 10) != 0){
```
Comment (piyalbasu) [inline packages/demo-wallet-server/src/index.ts:31]: NIT: this isn't technically wrong as these are both numbers, but it's good JS practice to always use`!==` (even if in this case it's unnecessary): https://bytearcher.com/articles/equality-comparison-operator-javascript/

File: packages/demo-wallet-shared/methods/sep10Auth/sign.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  secretKey: string;
+  walletBackendEndpoint: string;
+}) => {
+  log.instruction({
+    title:
+      "We’ve received a challenge transaction from the server that we need the sending anchor to sign with their Stellar private key",
+  });
+  
+  for (const op of challengeTransaction.operations) {
+    if (op.type == "manageData" && op.name == "client_domain") {
```
Comment (piyalbasu) [inline packages/demo-wallet-shared/methods/sep10Auth/sign.ts:21]: again, not wrong to use `==`, but good convention to use `===`

File: packages/demo-wallet-shared/methods/sep10Auth/sign.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      // The anchor server supports client attribution, get a signature from the demo wallet backend server
+      log.instruction({title: "anchor supports SEP-10 client attribution, requesting signature from demo wallet backend..."});
+      const params = {
+        transaction: challengeTransaction.toEnvelope().toXDR("base64"),
+        network_passphrase: networkPassphrase
+      };
+      log.request({ title: "POST `/sign`", body: params });
+      const urlParams = new URLSearchParams(params);
+      const walletBackendSignEndpoint = walletBackendEndpoint + "/sign";
```
Comment (piyalbasu) [inline packages/demo-wallet-shared/methods/sep10Auth/sign.ts:30]: let's use string concatenation here:
`${walletBackendEndpoint }/sign`

**Review by piyalbasu — APPROVED** [review]

**Review by stfung77 — APPROVED** [review]

**Pull request merged**